### PR TITLE
add modal for back, cancel action in import form

### PR DESCRIPTION
**Pull request by sahil143**
## Fixes 

https://issues.redhat.com/browse/RHTAPBUGS-258

## Description

If the user tries to leave the import form, confirmation modal should be shown

## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

image

## Browser conformance: 

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/ImportForm/GitImportForm.tsx [hunk 1]
```diff
@@ -149,13 +154,16 @@ const GitImportForm: React.FunctionComponent<GitImportFormProps> = ({
...
+      if (leave) {
+        setReviewMode(false);
+      }
+    });
   };
 
   return (
     <Formik
       onSubmit={reviewMode ? handleSubmit : handleNext}
-      onReset={handleReset}
```
Comment (rohitkrai03) [inline src/components/ImportForm/GitImportForm.tsx:158]: Why change sending reset handler from formik to directly?

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

File: src/components/ImportForm/GitImportForm.tsx (hunk 1 above)
Comment (sahil143) [reply src/components/ImportForm/GitImportForm.tsx:158]: because it would have reset the values of form before confirmation from the modal.

**Review by rottencandy — APPROVED** [review]
/lgtm

File: src/components/ImportForm/GitImportForm.tsx [hunk 2]
```diff
@@ -129,7 +130,11 @@ const GitImportForm: React.FunctionComponent<GitImportFormProps> = ({
 
   const handleReset = () => {
     track(TrackEvents.ButtonClicked, { link_name: 'import-leave', workspace });
-    navigate(-1);
+    showModal<{ leave: boolean }>(createCloseImportFormModal()).closed.then(({ leave }) => {
```
Comment (rohitkrai03) [inline src/components/ImportForm/GitImportForm.tsx:133]: I think before showing the modal we should also check if the form is dirty or not and only show the modal if it is.

**Review by rohitkrai03 — CHANGES_REQUESTED** [review]

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Pull request merged**